helpers/helpers.go

```diff
@@ -153,6 +156,70 @@ func GetIncludedBundles(filename string) ([]string, error) {
 	return includes, nil
 }
 
+// UnpackFile unpacks a .tar or .tar.gz/.tgz file to a given directory
+// Should be roughly equivalent to "tar -x[z]f file -C dest"
+func UnpackFile(file string, dest string) error {
+	fr, err := os.Open(file)
+	defer fr.Close()
```

Comment:
This and other defer statements must happen after the error handling. When error is returned, many functions return nil as the other value.

Reply:
> This and other defer statements must happen after the error handling.

I thought about this and found examples of both on the internet. Why do you say they need to be after? Wouldn't putting this after the `if err != nil` block mean the file never gets closed in an error state?

>When error is returned, many functions return nil as the other value.

Are you talking about in `AddBundles` where I return `bundleAddCount` instead of 0 (the nil of ints)? I did this because, at the point the error occured, `bundleAddCount` is actually a valid representation of how many bundles were added by the time the error occurred. If there's somewhere else I returned an odd value, I'm not aware of it.

Reply:
> Wouldn't putting this after the if err != nil block mean the file never gets closed in an error state?

No. The idiom is for functions to return "nil, err". This is true for the functions we are looking at in this PR. If the file wasn't successfully open, you don't need to close it, since there is no file in the first place.

The commit needs to be fixed.

> Are you talking about in AddBundles where I return bundleAddCount instead of 0 (the nil of ints)?

No. It was a bad attempt to explain the above. Sometimes returning "empty" containers is useful for the calling code, but for things like "give me a 'resource'", you usually return nil and an error.

Reply:
Agree 100% with @cmarcelo , the commit does need to be fixed.

It is more usual to say "the zero value of Something", than "the nil of Something". Go has an idiom that you try and make the zero value of Something be something that is usable.